File: git-depart/git-depart.php
<?php
declare(strict_types = 1);

system('git commit --verbose');

if (is_file('config.toml')) {
   $dir_o = new RecursiveDirectoryIterator('docs');
   $iter_o = new RecursiveIteratorIterator($dir_o);

   echo "UNLINK\n";
   foreach ($iter_o as $info_o) {
      if ($info_o->isFile()) {
         unlink($info_o->getPathname());
      }
   }

   echo "HUGO\n";
   system('hugo');
   echo "GIT ADD\n";
   system('git add .');
   echo "GIT COMMIT\n";
   system('git commit --amend');
}

echo "git push\n";

File: git-depart/git-depart.go
package main

import (
   "github.com/89z/x"
   "os"
)

func main() {
   e := x.System("git", "commit", "--verbose")
   x.Check(e)
   if x.IsFile("config.toml") {
      println(x.ColorGreen("remove docs"))
      os.RemoveAll("docs")
      println(x.ColorGreen("hugo"))
      e = x.System("hugo")
      x.Check(e)
      println(x.ColorGreen("git add"))
      e = x.System("git", "add", ".")
      x.Check(e)
      println(x.ColorGreen("git commit"))
      e = x.System("git", "commit", "--amend")
      x.Check(e)
   }
   println(x.ColorGreen("git push"))
}
